client/header/transient-notices/transient-notice.js

```diff
@@ -0,0 +1,102 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+import { Component } from '@wordpress/element';
+import { CSSTransition } from 'react-transition-group';
+import { noop } from 'lodash';
+import { Notice } from '@wordpress/components';
+import PropTypes from 'prop-types';
+
+class TransientNotice extends Component {
+	constructor( props ) {
+		super( props );
+
+		this.state = {
+			visible: false,
+			timeout: null,
+		};
+	}
+
+	componentDidMount() {
+		const exitTime = this.props.exitTime;
+		const timeout = setTimeout(
+			() => {
+				this.setState( { visible: false } );
+			},
+			exitTime,
+			name,
+			exitTime
+		);
+		/* eslint-disable react/no-did-mount-set-state */
+		this.setState( { visible: true, timeout } );
+		/* eslint-enable react/no-did-mount-set-state */
+	}
+
+	componentWillUnmount() {
+		clearTimeout( this.state.timeout );
+	}
+
+	render() {
+		const { actions, className, isDismissible, message, onRemove, status } = this.props;
+		const classes = classnames( 'woocommerce-transient-notice', className );
+
+		return (
+			<CSSTransition in={ this.state.visible } timeout={ 300 } classNames="slide">
+				<div className={ classes } role="alert">
+					<Notice
+						status={ status }
+						isDismissible={ isDismissible }
+						onRemove={ onRemove }
+						actions={ actions }
+					>
+						{ message }
```

Comment:
I recall us discussing somewhere ( can't quite remember where ) some of the a11y implications of these toast-style notifications. I seem to think we talked about using [speak](https://github.com/WordPress/gutenberg/blob/master/packages/a11y/src/index.js#L33) here but /cc @ryelle in case you might have some thoughts on how best to handle this.

Reply:
Interesting.  Any benefit to using `speak()` over `role="alert"` here @ryelle ?

Reply:
`speak` uses a container that has slightly different semantics – `role="alert"` will interrupt the screen reader as soon as it changes, while with `speak` you can control whether it should "politely" interrupt (waiting for the SR to finish reading the current action, generally preferred), or "assertively" interrupt (same behavior as `role="alert"`).

You might also want to use it to give slightly different content to a screen reader– for example, a lot of "toast-style" notices are hard to navigate focus to, so while sighted users can see and mouse to a button/link, maybe we just want to read a string message to a screen reader user (with the usual caveat that we shouldn't be showing very different content to SR users).

Reply:
Thanks for that explanation, @ryelle.  Added in `speak` in favor of `role="alert"` here.